pkg/build/ignore/docker/dockign.go

```diff
@@ -0,0 +1,116 @@
+package docker
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+type DOCKIGN struct {
+	Config *api.Config
+}
+
+func New(req *api.Config) *DOCKIGN {
+	d := &DOCKIGN{
+		Config: req,
+	}
+	return d
+}
+
+func (b *DOCKIGN) Ignore(config *api.Config) {
```

Comment:
So you ack'ed the existence of an interface below.

On the api.Config struct passed in:  I did this because
a) this makes Ignore/Ignorer looking consistent with Build/Builder, Prepare/Preparer, and their associated new Methods
b) if later enhancements call for more information needed off of api.Config, we already have the entire object, and don't need to change the callers to Ignore to pass in more info.  Since we are passing a pointer to the struct, it should not be expensive as a parameter.

On splitting the Ignore method into get the list, remove files from the list:
a) I failed to check in my unit test :-), but it was not terribly difficult 
b) otherwise though, I contemplated the split you described, but the delete part is so trivial in the end from a lines of code perspective, I did not see the value and putting another frame on the stack.

All this said ... let's circle back here after we cross off some of the other items from the review (including getting my unit test checked in this time).